Tasks/VsTestV2/runvstest.ts

```diff
@@ -27,38 +29,70 @@ if (osPlat !== 'win32') {
         }
         const serverBasedRun = isServerBasedRun();
         inputParser.setIsServerBasedRun(serverBasedRun);
+
         if (serverBasedRun) {
             ci.publishEvent({
                 runmode: 'distributedtest', parallelism: tl.getVariable('System.ParallelExecutionType'),
                 testtype: tl.getInput('testSelector')
             });
-
             console.log(tl.loc('distributedTestWorkflow'));
             console.log('======================================================');
             const inputDataContract = inputParser.parseInputsForDistributedTestRun();
             console.log('======================================================');
-
             const test = new distributedTest.DistributedTest(inputDataContract);
             test.runDistributedTest();
         } else {
             ci.publishEvent({ runmode: 'nondistributed' });
             console.log(tl.loc('nonDistributedTestWorkflow'));
             console.log('======================================================');
             const inputDataContract = inputParser.parseInputsForNonDistributedTestRun();
+            if (enableApiExecution || (inputDataContract.ExecutionSettings
+                && inputDataContract.ExecutionSettings.RerunSettings
+                && inputDataContract.ExecutionSettings.RerunSettings.RerunFailedTests)) {
+                console.log('================== API Execution =====================');
+                const test = new nondistributedtest.NonDistributedTest(inputDataContract);
+                test.runNonDistributedTest();
+            } else {
+                localtest.startTest();
+            }
             console.log('======================================================');
-
-            const test = new nondistributedtest.NonDistributedTest(inputDataContract);
-            test.runNonDistributedTest();
         }
     } catch (error) {
         tl.setResult(tl.TaskResult.Failed, error);
         taskProps.result = error;
-    } finally {
+    }
+    finally {
         taskProps.state = 'completed';
         ci.publishEvent(taskProps);
     }
 }
 
+function getFeatureFlag(collectionUri: string, featureFlag: string, token: string): Promise<boolean> {
+    let state = false;
+    const options = {
+        url: collectionUri + '/_apis/FeatureFlags/' + featureFlag,
```

Comment:
>FeatureFlags [](start = 37, length = 12)

general question. are these apis documented?

Reply:
answer is yes

---
In reply to: [202637953](https://github.com/Microsoft/vsts-tasks/pull/7669#discussion_r202637953) [](ancestors = 202637953)